heron/packing/src/java/com/twitter/heron/packing/Container.java

```diff
@@ -59,14 +67,48 @@ private boolean hasSpace(long ram, double cpuCores, long disk) {
    *
    * @return true if the instance can be added to the container, false otherwise
    */
-  public boolean add(Resource resource) {
+  public boolean add(Resource resource, InstanceId instanceId) {
     if (this.hasSpace(resource.getRam(), resource.getCpu(), resource.getDisk())) {
       usedRam += resource.getRam();
       usedCpuCores += resource.getCpu();
       usedDisk += resource.getDisk();
+      instanceIds.add(instanceId);
       return true;
     } else {
       return false;
     }
   }
+
+  /**
+   * Remove an instance of a particular component from a container and update its
+   * corresponding resources.
+   *
+   * @return instanceId if the instance is removed the container.
+   * Return void if an instance is not found
+   */
+  public Optional<InstanceId> removeAnyInstanceOfComponent(Resource resource, String component) {
+    Optional<InstanceId> instanceId = getAnyInstanceOfComponent(component);
+    if (instanceId.isPresent()) {
+      usedRam -= resource.getRam();
```

Comment:
Feel free to disregard this comment, as we don't follow this standard in many places.